DylibBinder/XmlGenerator.cs

```diff
@@ -0,0 +1,289 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml;
+using SwiftReflector.SwiftXmlReflection;
+
+namespace DylibBinder {
+	sealed internal class XmlGenerator : IDisposable {
+		XmlWriter writer;
+		private bool _disposed = false;
+
+		public XmlGenerator (DBTopLevel dBTopLevel, string outputPath)
```

Comment:
While this works, I get twitchy when I see all the work being done in the constructor _and_ when the class is `IDisposable`.  In looking at your code, it looks like it doesn't need to be.
You could instead structure the class like this:

```
sealed internal class XmlGenerator {
    string outputPath;
    XmlWriter writer;

    public XmlGenerator (string outputPath) {
        this.outputPath = outputPath; // throw on null kthx
    }

    public void WriteDBToFile (DBTopLevel dBTopLevel) {
        using (writer = CreateWriter (outputPath)) {
            WriteIntro (dBTopLevel);
            WriteModule (dBTopLevel);
            CloseWriter ();
        }
    }
// etc
}
```

When when you use it, you can do:
```
var generator = new XmlGenerator (outputPath);
generator.WriteDBToFile (db);
```
Which makes the flow clear for what you're doing:
I'm making an XmlGenerator, then I use it to write the the database. In addition, the writer, which is a transient resource, gets managed where it gets used rather than shoving it to the client. Another way of looking at it is to think of the implementation of writing as a mystery to the caller (as it should be). I call a method and it just works, I don't care how it gets done. When you make the class disposable, you're pushing the implementation details out to the caller who now needs to care about an implementation detail.

Now, this is just one way of looking at this from an API point of view. There are several other ways that you can do this.

For example, maybe the class holds onto an instance of the DB and you pass in the path to the method that does the work.

Another way to do this instead is to make the constructor (above) private and public a static method:

```
public static void WriteDBToFile (DBTopLevel db, string outputPath) {
    var generator = new XmlGenerator (outputPath);
    generator.WriteDBToFile (db);
}
```
The the callers do `XmlGenerator.WriteDBToFile (db, outputPath);`

Reply:
Okay I made those changes, thank you!